elscreen-buffer-list.el

```diff
@@ -0,0 +1,83 @@
+;;; http://www.emacswiki.org/emacs/elscreen-buffer-list.el
+;;; From http://www.emacswiki.org/emacs/ElscreenSeparateBufferLists
+
+(provide 'elscreen-buffer-list)
+(require 'elscreen)
+
+;; From escreen.el
+(defun buffer-dead-p (buffer)
+  (not (static-if (fboundp 'buffer-live-p)
+           (buffer-live-p buffer)
+         ;; Emacs 18 doesn't have buffer-live-p.
+         ;; Killed buffers have no names.
+         (buffer-name buffer))))
+
+;; From escreen.el
+(defun reorder-buffer-list (new-list)
+  "Set buffers in NEW-LIST to be the most recently used, in order."
+  (when new-list
+    (let (firstbuf buf)
+      (while new-list
+        (setq buf (car new-list))
+        (when (stringp buf)
+          (setq buf (get-buffer buf)))
+        (unless (buffer-dead-p buf)
+          (bury-buffer buf)
+          (unless firstbuf
+            (setq firstbuf buf)))
+        (setq new-list (cdr new-list)))
+      (setq new-list (buffer-list))
+      (while (not (eq (car new-list) firstbuf))
+        (bury-buffer (car new-list))
+        (setq new-list (cdr new-list))))))
+
+
+(defun elscreen-get-buffer-list (screen)
+  "Return buffer-list of SCREEN."
+  (let ((screen-property (elscreen-get-screen-property screen)))
+    (get-alist 'buffer-list screen-property)))
```

Comment:
This line doesn't work in emacs 24.4 w/ the current version of elscreen. Older versions of elscreen ship with:

```elisp
(defun get-alist (key alist)
  (cdr (assoc key alist)))
```